### PR TITLE
Variable Inlining [2/n]: Field Variable Inlining

### PR DESCRIPTION
**Pull request by Adarsh-NP**
This PR aims to solve variable inlining for field variables of a class. Please refer to the tests to understand the various cases it solves.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #459
* __->__ #458

### CODE REVIEW

File: src/cleanup_rules/swift/edges.toml [hunk 1]
```diff
@@ -18,6 +18,24 @@ scope = "Function"
 from = "delete_variable_declaration"
 to = ["replace_identifier_with_value", "delete_parent_assignment"]
 
+[[edges]]
+scope = "Class"
+from = "delete_field_initialisation_init"
+to = [
+  "replace_self_identifier_with_value",
+  "replace_identifier_with_value",
+  "delete_field_declaration",
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/edges.toml:27]: why not delete the assignments too?

File: src/cleanup_rules/swift/rules.toml [hunk 2]
```diff
@@ -609,6 +609,14 @@ queries = [
         (#eq? @var "@avariable")
         (#not-eq? @val "@avalue")
     )""",
+
+  # skip the rule if the variable is one of the parameters of some function
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/rules.toml:613]:
```suggestion
  # skip the rule if the variable is a parameter of a function
```

File: src/cleanup_rules/swift/rules.toml [hunk 3]
```diff
@@ -649,14 +654,201 @@ queries = [
...
+    (#eq? @avariable "@hvariable")
+)"""
+replace_node = "property_declaration"
+replace = ""
+holes = ["hvariable"]
+is_seed_rule = false
+
+[[rules.constraints]]
+matcher = "(class_declaration) @ccd"
+# rule to replace field variable used as self.x subject to the mentioned constraints
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/rules.toml:684]: this comment could be moved above l.667?

File: src/cleanup_rules/swift/rules.toml [hunk 4]
```diff
@@ -649,14 +654,201 @@ queries = [
...
+)"""
+replace_node = "rep"
+replace = "@hvalue"
+holes = ["hvariable", "hvalue"]
+is_seed_rule = false
+
+[[rules.constraints]]
+matcher = "(class_declaration) @cd"
+queries = [
+  # skip the rule if the variable is one of the parameters of some function
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/rules.toml:704]:
```suggestion
  # skip the rule if the variable is a parameter of a function
```

File: src/cleanup_rules/swift/rules.toml [hunk 5]
```diff
@@ -649,14 +654,201 @@ queries = [
...
+    )""",
+  # skip the rule if the variable declaration exists
+  """(
+        (property_declaration
+            name: (pattern
+                bound_identifier: (simple_identifier) @variable
+            )
+            value: (boolean_literal) @value
+        )
+        (#eq? @variable "@identifier")
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/rules.toml:719]: do we not want to put something like `(#not-eq/eq @value @hvalue)` ?

File: src/cleanup_rules/swift/edges.toml (hunk 1 above)
Comment (ketkarameya) [reply src/cleanup_rules/swift/edges.toml:27]: Also can we abstract these rules to groups ? And have edges just between the groups?

**Review by ketkarameya — CHANGES_REQUESTED** [review]

File: src/cleanup_rules/swift/rules.toml (hunk 3 above)
Comment (Adarsh-NP) [reply src/cleanup_rules/swift/rules.toml:686]: Can't be, different rules. But added a comment there too.

**Review by ketkarameya — APPROVED** [review]

**Pull request merged**